scripts/data_catalog.py

```
@@ -0,0 +1,161 @@
+        files = manager.list_files(source)
+        for file in files:
+            info = get_file_info(file)
+            info['source'] = source
+            results.append(info)
+    except ValueError:
+        pass
+    
+    return results
+
+
+def main():
```

Ruff (0.12.2)

`main` is too complex (11 > 10)

(C901)